jolt-core/src/jolt/vm/registers.rs

```diff
@@ -172,17 +174,21 @@ impl<F: JoltField, ProofTranscript: Transcript> RegistersTwistProof<F, ProofTran
 }
 
 impl<F: JoltField, ProofTranscript: Transcript> ReadWriteCheckingProof<F, ProofTranscript> {
-    #[tracing::instrument(skip_all, name = "ReadWriteCheckingProof::prove")]
-    pub fn prove(
-        trace: &[RV32IMCycle],
-        r: Vec<F>,
-        r_prime: Vec<F>,
+    #[tracing::instrument(skip_all, name = "ReadWriteCheckingProof::prove_with_array")]
+    pub fn prove_from_array(
+        write_addresses: Vec<usize>,
+        read_addresses: [Vec<usize>; 2],
+        read_values_1: Vec<u64>,
+        read_values_2: Vec<u64>,
+        write_values: Vec<u64>,
+        write_increments: Vec<i128>,
+        r: &[F],
+        r_prime: &[F],
         transcript: &mut ProofTranscript,
     ) -> (ReadWriteCheckingProof<F, ProofTranscript>, Vec<F>, Vec<F>) {
-        const DEGREE: usize = 3;
-        const K: usize = REGISTER_COUNT as usize;
```

Comment:
I would prefer to keep K a constant so that we can use arrays instead of vectors. iirc the performance difference is non-negligible